### PR TITLE
Update .gitignore

### PR DESCRIPTION
**Pull request by beutlich**
This adds more files to the .gitignore file

* Ignore VS upgrade files (when playing with VS compilers locally)
* Ignore GCC or MSVC libs (when building locally)
* Ignore files for generation of release notes
* Ignore tidy.dll (when running CI locally)

### CODE REVIEW

**Review by sjoelund — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,38 +1,46 @@
...
+.libs
+Modelica/Resources/BuildProjects/VisualStudio2005/*.filters
+Modelica/Resources/BuildProjects/VisualStudio2005/*.vcxproj
+Modelica/Resources/BuildProjects/autotools/AUTHORS
+Modelica/Resources/BuildProjects/autotools/COPYING
+Modelica/Resources/BuildProjects/autotools/ChangeLog
+Modelica/Resources/BuildProjects/autotools/INSTALL
+Modelica/Resources/BuildProjects/autotools/Makefile*
+Modelica/Resources/BuildProjects/autotools/NEWS
+Modelica/Resources/BuildProjects/autotools/compile
```
Comment (dietmarw) [inline .gitignore:23]: Why is this necessary. Why would one add all those files to the repo to begin with?

Comment (beutlich) [reply .gitignore:23]: It's already there. Just resorted alphabetically.

Comment (dietmarw) [reply .gitignore:23]: Why not removing it then. Seems to make the .gitignore unnecessary bloated.

Comment (beutlich) [reply .gitignore:23]: I guess @sjoelund added them by purpose in #2933.

Comment (beutlich) [reply .gitignore:23]: Also, it does not hurt ignoring them.

Comment (sjoelund) [reply .gitignore:23]: They are required by the automake stuff. The script generates them.

Comment (dietmarw) [reply .gitignore:23]: OK, I see.

**Review by dietmarw — APPROVED** [review]
